src/client/credentials_json_client.ts

```diff
@@ -0,0 +1,110 @@
+'use strict';
+
+import { createSign } from 'crypto';
+import { AuthClient } from './auth_client';
+import { toBase64, fromBase64, writeSecureFile } from '../utils';
+
+/**
+ * Available options to create the WorkloadIdentity client.
+ *
+ * @param projectID User-supplied value for project ID. If not provided, the
+ * project ID is extracted from the credentials JSON.
+ * @param credentialsJSON Raw JSON credentials blob.
+ */
+interface CredentialsJSONClientOptions {
+  projectID?: string;
+  credentialsJSON: string;
+}
+
+export class CredentialsJSONClient implements AuthClient {
+  readonly #projectID: string;
+  readonly #credentials: Record<string, string>;
+
+  constructor(opts: CredentialsJSONClientOptions) {
+    this.#credentials = this.parseServiceAccountKeyJSON(opts.credentialsJSON);
+    this.#projectID = opts.projectID || this.#credentials['project_id'];
+  }
+
+  /**
+   * parseServiceAccountKeyJSON attempts to parse the given string as a service
+   * account key JSON. It handles if the string is base64-encoded.
+   */
+  parseServiceAccountKeyJSON(str: string): Record<string, string> {
```

Comment:
Should we validate this into a `ServiceAccountKey` type? This way we can fail fast if malformed.

Reply:
Done. I only included the fields we need to be successful though. I want to avoid taking super hard dependencies on the format of a service account JSON.